code/lib/core-server/src/utils/StoryIndexGenerator.test.ts

```diff
@@ -0,0 +1,1741 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+/// <reference types="@types/jest" />;
+
+/**
+ * @jest-environment node
+ */
+
+import path from 'path';
+import fs from 'fs-extra';
+import { normalizeStoriesEntry } from '@storybook/core-common';
+import type { Indexer, NormalizedStoriesSpecifier, StoryIndexEntry } from '@storybook/types';
+import { loadCsf, getStorySortParameter } from '@storybook/csf-tools';
+import { toId } from '@storybook/csf';
+import { logger, once } from '@storybook/node-logger';
+
+import type { StoryIndexGeneratorOptions } from './StoryIndexGenerator';
+import { StoryIndexGenerator } from './StoryIndexGenerator';
+
+jest.mock('@storybook/csf-tools');
+jest.mock('@storybook/csf', () => {
+  const csf = jest.requireActual('@storybook/csf');
+  return {
+    ...csf,
+    toId: jest.fn(csf.toId),
+  };
+});
+
+jest.mock('@storybook/node-logger');
+
+const toIdMock = toId as jest.Mock<ReturnType<typeof toId>>;
+const loadCsfMock = loadCsf as jest.Mock<ReturnType<typeof loadCsf>>;
+const getStorySortParameterMock = getStorySortParameter as jest.Mock<
+  ReturnType<typeof getStorySortParameter>
+>;
+
```

Comment:
these two indexers below are very likely how they will be implemented for real in a follow up PR